src/tricicl/storage/remote_storage.py

```diff
@@ -0,0 +1,84 @@
+import shutil
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path, PurePath
+from tempfile import TemporaryDirectory
+from typing import List
+
+from tricicl.constants import PROJECT_DIR
+
+
+@dataclass
+class SyncPath:
+    remote: PurePath
+    local: Path
+
+    @staticmethod
+    def from_local(local_path: Path) -> "SyncPath":
+        return SyncPath(remote=local_path.relative_to(PROJECT_DIR), local=local_path)
+
+    @staticmethod
+    def from_remote(remote_path: PurePath) -> "SyncPath":
+        return SyncPath(remote=remote_path, local=PROJECT_DIR / remote_path)
+
+    @staticmethod
+    def from_str(rel_path: str) -> "SyncPath":
+        return SyncPath.from_local(PROJECT_DIR / rel_path)
+
+
+class RemoteStorageABC(ABC):
+    def download_dataset(self, name: str):
+        self.download_zipped_directory_if_missing(get_dataset_sync_path(name))
+
+    def upload_dataset(self, name: str):
+        self.upload_zipped_directory(get_dataset_sync_path(name))
```

Comment:
This is convenient because from gcloud downloading from a bucket is way faster than from a url. I guess it should be similar in AWS, if you want to benefit from those method (plus aggregation from S3 storage), you simply need to inherit `RemoteStorageABC` and implement the 3 basic abstract methods